packages/nextra-theme-blog/src/mdx-theme.tsx

```diff
@@ -65,10 +66,11 @@ const A = ({ children, href = '', ...props }: ComponentProps<'a'>) => {
       </a>
     )
   }
+  const ComponentToUse = ID_HREF_REGEX.test(href) ? 'a' : Link
```

Comment:
Since regex contains only 1 character and start with `^` character, we can simplify it to, wdyt?
```suggestion
  const ComponentToUse = href.startsWith('#') ? 'a' : Link
```

Reply:
Keeping it simple is a great idea! I referred to the approach used in `EXTERNAL_HREF_REGEX` to avoid receiving props other than strings, which could lead to type errors. Perhaps I was overthinking it.

Reply:
Yeah, I got it, you stayed with consistentsy with the current code 😊👍